packages/zero-pg/src/push-processor.pg-test.ts

```diff
@@ -304,6 +312,209 @@ test('mutators with and without namespaces', async () => {
+    );
+
+    mutators = {
+      postCommit: {
+        noError: tx => {
+          tx.after(noErrorMock);
+          return Promise.resolve();
+        },
+        taskError: tx => {
+          tx.after(taskErrorMock);
+          return Promise.resolve();
+        },
+        mutationError: tx => {
+          tx.after(mutationErrorMock);
+          return Promise.reject(new Error('mutator error'));
+        },
+        asyncResolution: tx => {
+          tx.after(asyncResolutionMock);
+          return Promise.resolve();
+        },
+      },
+    };
+  });
+
+  afterEach(() => {
+    consoleErrorMock.mockReset();
+  });
+
+  test('tasks execute if mutator succeeds', async () => {
+    const processor = new PushProcessor(
+      new ZQLDatabaseProvider(new ZQLPostgresJSAdapter(pg), schema),
+    );
+
+    await processor.process(
+      mutators,
+      params,
+      makePushBatch([
+        [1, customMutatorKey('postCommit', 'noError')],
+        [2, customMutatorKey('postCommit', 'taskError')],
+      ]),
+    );
+
+    expect(noErrorMock).toHaveBeenCalledTimes(1);
+    expect(taskErrorMock).toHaveBeenCalledTimes(1);
+  });
+
+  test('tasks do not execute if mutator throws', async () => {
+    const processor = new PushProcessor(
+      new ZQLDatabaseProvider(new ZQLPostgresJSAdapter(pg), schema),
+    );
+
+    await processor.process(
+      mutators,
+      params,
+      makePushBatch([
+        [1, customMutatorKey('postCommit', 'noError')],
+        [2, customMutatorKey('postCommit', 'mutationError')],
+      ]),
+    );
+
+    expect(noErrorMock).toHaveBeenCalledTimes(1);
+    expect(mutationErrorMock).toHaveBeenCalledTimes(0);
+  });
+
+  test('tasks throws are sent to the logger', async () => {
+    const processor = new PushProcessor(
+      new ZQLDatabaseProvider(new ZQLPostgresJSAdapter(pg), schema),
+    );
+
+    await processor.process(
+      mutators,
+      params,
+      makePushBatch([
+        [1, customMutatorKey('postCommit', 'noError')],
+        [2, customMutatorKey('postCommit', 'taskError')],
+      ]),
+    );
+
+    expect(noErrorMock).toHaveBeenCalledTimes(1);
+    expect(taskErrorMock).toHaveBeenCalledTimes(1);
+
+    expect(consoleErrorMock).toHaveBeenCalledTimes(1);
+    expect(consoleErrorMock.mock.calls[0][0]).toEqual('error');
+    expect(consoleErrorMock.mock.calls[0][2]).toEqual(taskException);
+  });
+
+  test('.process() waits for tasks to settle by default', async () => {
+    const processor = new PushProcessor(
+      new ZQLDatabaseProvider(new ZQLPostgresJSAdapter(pg), schema),
+    );
+
+    const processPromise = processor.process(
+      mutators,
+      params,
+      makePushBatch([[1, customMutatorKey('postCommit', 'asyncResolution')]]),
+    );
+
+    let processCompleted = false;
+    void processPromise.then(() => {
+      processCompleted = true;
+    });
+
+    // Wait for the full mutation flow to complete w/ PG.
+    for (let i = 0; i < 10; i++) {
+      await sleep(10);
+    }
```

Comment:
don't love the timing stuff in this and subsequent test cases but i didn't see a better way to avoid flaking, if there is a nicer pattern for this from other tests in the codebase point it my way and I will improve it!